scripts/paypal/payment-reconciliator.ts

```diff
@@ -530,43 +467,108 @@ const findMissingPaypalTransactions = async (_, commander) => {
       totalPages = <number>response['totalPages'];
 
       // Make sure all transactions exist in the ledger
-      for (const paypalTransaction of <Record<string, unknown>[]>response['transaction_details']) {
+      for (const paypalTransaction of <PaypalTransactionSearchResult['transaction_details']>(
+        response['transaction_details']
+      )) {
+        console.log(
+          `Checking transaction ${paypalTransaction['transaction_info']['transaction_id']}...`,
+          paypalTransaction,
+        );
         const transactionInfo = paypalTransaction['transaction_info'];
         const paypalTransactionId = <string>transactionInfo['transaction_id'];
-        const ledgerTransaction = await models.Transaction.findOne({
-          where: {
-            HostCollectiveId: host.id, // Pre-filter to make the query faster
-            data: {
-              [Op.or]: [{ capture: { id: paypalTransactionId } }, { paypalSale: { id: paypalTransactionId } }],
-            },
-          },
-        });
-
+        const ledgerTransaction = await findTransactionByPaypalId(paypalTransactionId, { HostCollectiveId: host.id });
         if (!ledgerTransaction) {
-          console.warn(`Missing PayPal transaction ${paypalTransactionId} in ledger`);
-          if (options['fix']) {
-            // Trigger the actual charge
+          const captureUrl = `payments/captures/${paypalTransactionId}`;
+          const captureDetails = (await paypalRequestV2(captureUrl, host, 'GET')) as PaypalCapture;
+          if (captureDetails.status !== 'COMPLETED') {
+            continue; // Make sure the capture is not pending
+          }
 
-            const captureDetails = await paypalRequestV2(`payments/captures/${paypalTransactionId}`, host, 'GET');
-            if (captureDetails.status !== 'COMPLETED') {
-              continue; // Make sure the capture is not pending
-            }
+          // Make sure it's happening on a subscription
+          console.warn(`Missing PayPal transaction ${paypalTransactionId} in ledger`);
+          if (transactionInfo['paypal_reference_id_type'] !== 'SUB') {
+            console.error(
+              `Found a missing transaction that is not a subscription: ${paypalTransactionId}`,
+              paypalTransaction,
+            );
+            throw new Error('This case is not supported'); // Shouldn't happen, just in case
```

Comment:
Shouldn't we `continue` here?

Reply:
I want to make sure we spot the error right away if this ever happens. And as mentioned, this script is probably going away (at least partially) in the near future